src/bases.jl

```diff
@@ -0,0 +1,595 @@
+hat!(M::Manifold, X, p, Xⁱ) = get_vector!(M, X, p, Xⁱ, DefaultBasis())
+
+"""
+    number_system(::AbstractBasis)
+
+The number system used as scalars in the given basis.
+"""
+number_system(::AbstractBasis{𝔽}) where {𝔽} = 𝔽
+
+function _show_basis_vector(io::IO, X; pre = "", head = "")
+    sX = sprint(show, "text/plain", X, context = io, sizehint = 0)
+    sX = replace(sX, '\n' => "\n$(pre)")
+    print(io, head, pre, sX)
+end
+function _show_basis_vector_range(io::IO, Ξ, range; pre = "", sym = "E")
+    for i in range
+        _show_basis_vector(io, Ξ[i]; pre = pre, head = "\n$(sym)$(i) =\n")
+    end
+    return nothing
+end
+function _show_basis_vector_range_noheader(io::IO, Ξ; max_vectors = 4, pre = "", sym = "E")
+    nv = length(Ξ)
+    if nv ≤ max_vectors
+        _show_basis_vector_range(io, Ξ, 1:nv; pre = "  ", sym = " E")
+    else
+        halfn = div(max_vectors, 2)
+        _show_basis_vector_range(io, Ξ, 1:halfn; pre = "  ", sym = " E")
+        print(io, "\n ⋮")
+        _show_basis_vector_range(io, Ξ, (nv-halfn+1):nv; pre = "  ", sym = " E")
+    end
+end
+
+function show(io::IO, ::DefaultOrthonormalBasis{𝔽}) where {𝔽}
+    print(io, "DefaultOrthonormalBasis($(𝔽))")
+end
+function show(io::IO, ::ProjectedOrthonormalBasis{method,𝔽}) where {method,𝔽}
+    print(io, "ProjectedOrthonormalBasis($(repr(method)), $(𝔽))")
+end
+function show(io::IO, mime::MIME"text/plain", onb::DiagonalizingOrthonormalBasis)
+    println(
+        io,
+        "DiagonalizingOrthonormalBasis with coordinates in $(number_system(onb)) and eigenvalue 0 in direction:",
+    )
+    sk = sprint(show, "text/plain", onb.frame_direction, context = io, sizehint = 0)
+    sk = replace(sk, '\n' => "\n ")
+    print(io, sk)
+end
+function show(
+    io::IO,
+    mime::MIME"text/plain",
+    B::CachedBasis{T,D,𝔽},
+) where {T<:AbstractBasis,D,𝔽}
+    vectors = _get_vectors(B)
+    nv = length(vectors)
+    print(
+        io,
+        "$(T()) with coordinates in $(number_system(B)) and $(nv) basis vector$(nv == 1 ? "" : "s"):",
+    )
+    _show_basis_vector_range_noheader(io, vectors; max_vectors = 4, pre = "  ", sym = " E")
+end
+function show(
+    io::IO,
+    mime::MIME"text/plain",
+    B::CachedBasis{T,D,𝔽},
+) where {T<:DiagonalizingOrthonormalBasis,D<:DiagonalizingBasisData,𝔽}
+    vectors = _get_vectors(B)
+    nv = length(vectors)
+    sk = sprint(show, "text/plain", T(B.data.frame_direction), context = io, sizehint = 0)
+    sk = replace(sk, '\n' => "\n ")
+    print(io, sk)
+    println(io, "\nand $(nv) basis vector$(nv == 1 ? "" : "s").")
+    print(io, "Basis vectors:")
+    _show_basis_vector_range_noheader(io, vectors; max_vectors = 4, pre = "  ", sym = " E")
+    println(io, "\nEigenvalues:")
+    sk = sprint(show, "text/plain", B.data.eigenvalues, context = io, sizehint = 0)
+    sk = replace(sk, '\n' => "\n ")
+    print(io, ' ', sk)
+end
+
+vee(M::Manifold, p, X) = get_coordinates(M, p, X, DefaultBasis())
```

Comment:
Wouldn't it cause problems that `vee` has a different default in `ManifoldsBase.jl`?

Reply:
Currently I am getting an overwrite warning, but this will replace the one in Base once tests here are running; then We can put bases and numbers over to base.

Reply:
I hope there are no failures caused by that warning, I have always resolved overwrite warnings before fixing bugs (I'm not sure though if it's strictly necessary).

Reply:
I am testing with the overwritten thing, I am quite sure that'll work. Most things even do work already.